src/main/java/forceitembattle/listener/Listeners.java

```diff
@@ -65,42 +71,117 @@ public void onPlayerQuit(PlayerQuitEvent playerQuitEvent) {
+        ItemStack itemStack = foundItemEvent.getFoundItem();
+
+        ForceItemBattle.getGamemanager().getScore().put(player.getUniqueId(), ForceItemBattle.getGamemanager().getScore().get(player.getUniqueId()) + 1);
+        ArrayList<ForceItem> mat = ForceItemBattle.getGamemanager().getItemList(player);
+        mat.add(new ForceItem(itemStack.getType(), ForceItemBattle.getTimer().formatSeconds(ForceItemBattle.getTimer().getTime()), foundItemEvent.isSkipped()));
+        ForceItemBattle.getGamemanager().getItemList().put(player.getUniqueId(), mat);
+        ForceItemBattle.getGamemanager().getCurrentMaterial().put(player.getUniqueId(), ForceItemBattle.getGamemanager().generateMaterial());
+        player.playSound(player.getLocation(), Sound.BLOCK_NOTE_BLOCK_BELL, 1, 1);
+
+        ArmorStand armorStand = (ArmorStand) player.getPassengers().get(0);
+        armorStand.getEquipment().setHelmet(new ItemStack(ForceItemBattle.getGamemanager().getCurrentMaterial(player)));
+
+        Bukkit.broadcastMessage("§a" + player.getName() + " §7" + (foundItemEvent.isSkipped() ? "skipped" : "found") + " §6" + WordUtils.capitalize(itemStack.getType().name().toLowerCase().replace("_", " ")));
+    }
+
     @EventHandler
     public void onClick(PlayerInteractEvent e) { // triggered if a joker is used
         if (!ForceItemBattle.getTimer().isRunning()) return;
         if (e.getPlayer().getInventory().getItemInMainHand().getType() != Material.BARRIER) return;
         if (!ForceItemBattle.getGamemanager().isPlayerInMaps(e.getPlayer())) return;
+        /*
         if (ForceItemBattle.getGamemanager().hasDelay(e.getPlayer())) {
             e.getPlayer().sendMessage(ChatColor.RED + "Please wait a second.");
             return;
         }
+        */
 
         if(e.getAction() == Action.RIGHT_CLICK_AIR) {
-            ItemStack stack = e.getPlayer().getInventory().getItem(e.getPlayer().getInventory().first(Material.BARRIER));
-            if (stack.getAmount() > 1) {
-                stack.setAmount(stack.getAmount() - 1);
-            } else {
-                stack.setType(Material.AIR);
-            }
-            Material mat;
-            if (ForceItemBattle.getInstance().getConfig().getBoolean("settings.isTeamGame")) {
-                /////////////////////////////////////// TEAMS ///////////////////////////////////////
-                mat = ForceItemBattle.getGamemanager().getMaterialTeamsFromPlayer(e.getPlayer());
-            } else {
-                mat = ForceItemBattle.getGamemanager().getCurrentMaterial(e.getPlayer());
+            if(e.getPlayer().getInventory().getItemInMainHand().getType() == Material.BARRIER) {
+                ItemStack stack = e.getPlayer().getInventory().getItem(e.getPlayer().getInventory().first(Material.BARRIER));
+                if (stack.getAmount() > 1) {
+                    stack.setAmount(stack.getAmount() - 1);
+                } else {
+                    stack.setType(Material.AIR);
+                }

```

Comment:
and instead of this just set it to the number of skips left after the current skip would be used